app/lib/account/backend.dart

```diff
@@ -258,8 +258,21 @@ class AccountBackend {
   Future<String> siteAuthCodeToAccessToken(String redirectUrl, String code) =>
       _authProvider.authCodeToAccessToken(redirectUrl, code);
 
-  /// Authenticates with bearer [token] and returns an `AuthenticatedUser`
-  /// object.
+  /// Verifies that the access token is matching the current active `User`.
+  ///
+  /// Throws [AuthenticationException] if token cannot be authenticated or the
+  /// OAuth userId differs from the current active user.
+  Future<void> verifyAccessToken(String accessToken) async {
```

Comment:
Can't we do a better name... `verifyAccessTokenOwnership(String, User owner)`?

Using `_authenticatedUser` feels a bit implicit. Or maybe it's just me, but this method doesn't verify an access token, it does something very different.

It sort of verifies ownership of the access token.

side note: The caller could just have authenticated with the access token in the first place.